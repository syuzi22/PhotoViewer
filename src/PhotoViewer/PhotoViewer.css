.viewer{
    display: flex;
    width: 100%;
    justify-content: center;
    align-items: center;

    img {
        width:80%;
        height:auto;
    }
}

