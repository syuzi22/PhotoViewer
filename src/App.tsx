import { useState } from "react";
import { PhotoViewer } from "./PhotoViewer/PhotoViewer";
import "./App.css";
import { ImageSelector } from "./ImageSelector/ImageSelector";
import { imageUrls } from "./utils.ts";

function App() {
  const [selectedImage, setSelectedImage] = useState(imageUrls[0]);

  const handleClick = (url: string) => {
    setSelectedImage(url);
  };

  return (
    <div className="photo-viewer">
      <h1>React Photo Viewer</h1>
      <PhotoViewer url={selectedImage} />
      <ImageSelector imageUrls={imageUrls} onClick={handleClick} selectedImage={selectedImage}/>
    </div>
  );
}

export default App;
