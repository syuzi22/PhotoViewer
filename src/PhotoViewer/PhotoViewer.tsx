import './PhotoViewer.css';

interface PhotoViewerProps {
  url: string;
}

export const PhotoViewer = ({ url }: PhotoViewerProps) => {
  return (
    <div className="viewer">
      <img src={url} data-testid="image" />
    </div>
  );
};
