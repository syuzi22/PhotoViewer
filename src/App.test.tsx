import { render,screen } from '@testing-library/react'
import { PhotoViewer } from './PhotoViewer/PhotoViewer';
import App from './App';
import { imageUrls } from "./utils";

test('renders React Photo Viewer text', () => {
    const { getByText } = render(<App />);
    const textElement = getByText(/React Photo Viewer/i);
    expect(textElement).toBeVisible();
});

test('renders React Photo Viewer image', () => {
    render(<PhotoViewer url='https://picsum.photos/id/600/1600/900.jpg'/>);
    const imageElement = screen.getByTestId('image') as HTMLImageElement;
    expect(imageElement).toBeVisible();
});

// A unit test to check our imageUrl generation code - 
// for me, this might check that the first link is what I expect and that it doesn’t include the ‘broken’ images.

test('getImageUrls generates expected images',() => {
    const url = imageUrls[0];
    expect(url).toBe("https://picsum.photos/id/600/1600/900.jpg");
});

/*test('getImageUrls to not include broken images',() => {
    const url = imageUrls[brokenImages[0]];
    expect(url).toBe(undefined);

});*/