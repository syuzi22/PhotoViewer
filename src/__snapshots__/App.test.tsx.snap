// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders photoviewer image correctly 1`] = `[Function]`;
